pkg/cluster/config/v1alpha3/types.go

```diff
@@ -82,3 +85,14 @@ const (
 	// WorkerRole identifies a node that hosts a Kubernetes worker
 	WorkerRole NodeRole = "worker"
 )
+
+// Network contains cluster wide network settings
+type Network struct {
+	// APIServerPort is the listen port on the host for the Kubernetes API Server
+	// Defaults to a random port on the host
+	APIServerPort int32 `json:"apiServerHostPort,omitempty"`
+	// APIServerListenAddress is the listen address on the host for the Kubernetes API Server
+	// This should be an IP address
+	// Defaults to 127.0.0.1
+	APIServerListenAddress string `json:"apiServerAddress,omitempty"`
```

Comment:
my preference would be to match the json tag names and the go field names.

also possibly use `Host` for both?
```
APIServerHostAddress `json:"apiServerHostAddress,omitempty"`
APIServerHostPort `json:"apiServerHostPort,omitempty"`
```

or use `Listen` for both.

but not strong opinions as long as the comments explain the fields.

here is the naming in kubeadm:
https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1#APIEndpoint

but that's a bit different as we try to map to the kube-apiserver CLI flags:
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

Reply:
+1 to making them match json, this field is a bit different as well because it's a host mapping to the actual one, the kubeadm config value should NOT match this value in most cases.

Reply:
done